src/FSharp.Stats/Fitting/GeneralisedLinearModel.fs

```diff
@@ -0,0 +1,668 @@
+    //         Vector.map2(fun y yhat -> (y-yhat)**2.) b linpred 
+    //         |> Vector.sum
+    //     let r2 = 1. - (rss / tss)
+    //     r2
+
+    let getGLMStatisticsModel (b:Vector<float>) (glmResult:GLMReturn) (family: GlmDistributionFamily) = 
+        let logLikelihood = getLogLikelihood b glmResult.mu
+        let deviance = GlmDistributionFamily.getFamilyResidualDeviance family b glmResult.mu 
+        let chi2 = getChi2 b glmResult.mu family
+        //let r2 = testR2 b (glmResult.mX * A)
+
+        {
+            LogLikelihood=logLikelihood
+            Deviance=deviance
+            PearsonChi2=chi2
+            //PseudoR2=0.
+        }
+
+    /// Calculates the standard errors for the coefficients in a generalized linear model.
+    /// The standard errors are calculated using the formula: sqrt(diagonal elements of (A^T * W * A)^-1)
+    /// where A is the design matrix, b is the response vector, and W is the weight vector.
+    let getStandardError (A: Matrix<float>) (b: Vector<float>) (W: Vector<float>) =
+        let At :Matrix<float> = Matrix.transpose A
+        let WMatrix = Matrix.diag W
+        let AtW = At * WMatrix
+        let AtWA :Matrix<float> = AtW*A
+        let AtWAInv = Algebra.LinearAlgebra.Inverse AtWA
+
+        let n = AtWAInv.NumRows
+        let m = Vector.length b
+        let stndErrors: Vector<float> = 
+            Vector.init n (fun v -> 
+                Matrix.get AtWAInv v v
+                |> fun x -> System.Math.Sqrt(x)
+            )
+        stndErrors
+
+    /// Calculates the Z-statistic for the coefficients in a generalized linear model.
+    /// The Z-statistic is calculated as the ratio of the coefficient estimate to its standard error.
+    let getZStatistic (mx: Vector<float>) (stndError: Vector<float>) = 
+        Vector.map2 (fun x y -> 
+            x/y
+        ) mx stndError
+
+    /// Calculates the p-value using the z-statistic.
+    /// The p-value is calculated as 2 * (1 - phi), where phi is the cumulative distribution function (CDF) of the standard normal distribution.
+    /// The z-statistic is a vector of values for which the p-value is calculated.
+    let getPearsonOfZ (zStatistic: Vector<float>) = 
+        Vector.map(fun x ->
+            let phi = Distributions.Continuous.Normal.CDF 0. 1. (abs(x))
+            let pValue = 2. * (1. - phi)
+            pValue
+        )zStatistic
+
+    let getGLMParameterStatistics (A:Matrix<float>) (b:Vector<float> ) (solved:GLMReturn) (names:string seq) =
+
+        let stndErrors = getStandardError A b solved.mu
+        let zStatistic = getZStatistic solved.mX stndErrors
+        let pValue = getPearsonOfZ zStatistic
+        Seq.init (Vector.length solved.mX) (fun i -> 
+            Seq.item i names,
+            {
+                Coefficient=solved.mX.[i]
+                StandardError=stndErrors.[i]
+                ZScore=zStatistic.[i]
+                PersonOfZ=pValue.[i]
+            }
+        )
+
+module QR =
```

Comment:
Is this the correct location for a function like this? Are these functions so specialized that nobody ever needs to use them in another context? if not, i'd suggest putting them in the managed LinearAlgebra module, otherwise you can also set the whole QR module here to internal

Reply:
Allrighty. You can then mark the whole specialized module as internal to avoid leaking any of that stuff in the public API.

Reply:
Doing that would also remove it from the testing I on the stepwise function. You can have a look and tell me if it is ok now.

Reply:
You can expose internals to other projects in the repo by adding this attribute in the project file:

```xml
<ItemGroup> 
  <InternalsVisibleTo Include="TestProject1" /> 
</ItemGroup>
```

Reply:
@kMutagene hey, I changed the code accordingly. Could you have another look if its better now?